interceptor.go

```diff
@@ -60,6 +68,14 @@ type Streamer func(ctx context.Context, desc *StreamDesc, cc *ClientConn, method
 //
 // StreamClientInterceptor may return a custom ClientStream to intercept all I/O
 // operations. The returned error must be compatible with the status package.
+//
+// A simple example that logs the method name when called and then invokes the
+// the stream.
+//
+//	func Interceptor(ctx context.Context, desc *StreamDesc, cc *ClientConn, method string, streamer Streamer, opts ...CallOption) (ClientStream, error) {
```

Comment:
I think the snippets here are not as useful as what they appear to be.
The streaming interceptors, (unlike the unary interceptors,) can be much more complex than the snippet.

We actually have interceptor examples, with client/server unary/streaming interceptors: https://github.com/grpc/grpc-go/tree/master/examples/features/interceptor

Maybe it makes more sense to link to the examples instead?